output/transform/merge_partition_sizes.go

```diff
@@ -0,0 +1,56 @@
+package transform
+
+import (
+	snapshot "github.com/pganalyze/collector/output/pganalyze_collector"
+	"github.com/pganalyze/collector/state"
+)
+
+func mergePartitionSizes(s snapshot.FullSnapshot, newState state.PersistedState, ts state.TransientState, databaseOidToIdx OidToIdx) snapshot.FullSnapshot {
+	for idx, info := range s.IndexInformations {
+		rel := s.RelationInformations[info.RelationIdx]
+		if !rel.HasParentRelation || rel.PartitionBoundary == "" {
+			continue
+		}
+		for parentIdx, pi := range s.IndexInformations {
+			if pi.RelationIdx != rel.ParentRelationIdx {
+				continue
+			}
+			if info.IndexType == pi.IndexType && info.IsUnique == pi.IsUnique && intArrayEqual(info.Columns, pi.Columns) {
+				s.IndexStatistics[parentIdx].SizeBytes += s.IndexStatistics[idx].SizeBytes
+				break
+			}
+		}
+	}
+
+	for idx, rel := range s.RelationInformations {
+		if !rel.HasParentRelation || rel.PartitionBoundary == "" {
+			continue
+		}
+		stat := s.RelationStatistics[idx]
+		parent := s.RelationStatistics[rel.ParentRelationIdx]
+		parent.NTupIns += stat.NTupIns
+		parent.NTupUpd += stat.NTupUpd
```

Comment:
Note: currently, Index Advisor is relying on this relation stats info.
Since the current parent table doesn't have the "writes" information (as these value being 0), it was hard (well, impossible) for the partitioned table to have a write optimized as a table default config as writes per min was always being 0.
With this change, this will change and maybe some partitioned tables will now have write optimized config to run the index advisor, so people may see some different index suggestions because of this.
I don't think it's a bad change (rather, the current behavior is inaccurate so it is a good change), but I think we want to keep this in mind and anticipate some potential confusion.